### ISSUE TITLE
Compatibility to ESP32C3 (Bootloop)

### ISSUE THREAD

**Issue by TheAndi** [fsaris/EspHome-AwoX-BLE-mesh-hub#111]
Thanks for the great work!

On a vanilla ESP32 the component with the given YAML works fine. 

But I wanted to use it on the small and cheap ESP32C3-Supermini. 
I only adopted the `board` in the YAML and flashed the C3, but it always results in a boot loop. 

Being quick with connecting to USB I could capture:

```
[20:01:04]ESP-ROM:esp32c3-api1-20210207
[20:01:04]Build:Feb  7 2021
[20:01:04]rst:0x7 (TG0WDT_SYS_RST),boot:0xd (SPI_FAST_FLASH_BOOT)
[20:01:04]Saved PC:0x40049a42
[20:01:04]SPIWP:0xee
[20:01:04]mode:QIO, clock div:1
[20:01:04]load:0x3fcd5810,len:0x1684
[20:01:04]ets_loader.c 78 
[20:01:05]ESP-ROM:esp32c3-api1-20210207
[20:01:05]Build:Feb  7 2021
[20:01:05]rst:0x7 (TG0WDT_SYS_RST),boot:0xd (SPI_FAST_FLASH_BOOT)
Terminal disconnected: NetworkError: The device has been lost.
```

The C3 should also support Bluetooth (LE). Is there something special to do, to make it work?

Thanks in advance!

**Comment by fsaris** [comment]
You could try without debug enabled.

But probably it's a memory issue.
Does the Bluetooth proxy work on the device?

**Comment by TheAndi** [comment]
Thanks for the reply. I can try that

Do you mean just the BT proxy from ESPHome?

**Comment by fsaris** [comment]
Yes, https://esphome.io/components/bluetooth_proxy.html

**Comment by TheAndi** [comment]
I did lots of testing... 
turned out It's nothing related with your component. I could not even get the bare bone config to run without boot loop.
I also tried a second module, but the cause was in the flash_mode (when using `esp-idf`), which is not well documented/known.


This how it's working. Maybe it helps someone:

```
esphome:
  name: XXX
  friendly_name: "XXX"
  platformio_options:
    board_build.flash_mode: dio
```

So I can confirm, that your module is **working with ESP32C3**, but I still have the same issues as mentioned [here](https://github.com/fsaris/EspHome-AwoX-BLE-mesh-hub/issues/113).
